google-cloud-pubsub/lib/google/cloud/pubsub/subscriber/timed_unary_buffer.rb

```diff
@@ -129,6 +154,60 @@ def stopped?
 
           private
 
+          def handle_failure error, ack_ids, ack_deadline_seconds = nil
+            ack_ids = (parse_error error) || ack_ids
+            perform_retry_async ack_ids, ack_deadline_seconds
+          end
+
+          def parse_error error
+            metadata = error.error_metadata
+            return if metadata.nil?
+            permanent_failures, temporary_failures = metadata.partition do |_, v|
+              v.include? PERMANENT_FAILURE
+            end.map(&:to_h)
+            handle_permanent_failures permanent_failures
+            return temporary_failures.keys.map(&:to_s) unless temporary_failures.empty?
```

Comment:
`return` is redundant here. (The `unless` guard already evaluates to `nil` if it fails.)